src/System.Numerics.Experimental/Half.cs

```diff
@@ -341,17 +353,358 @@ public override int GetHashCode()
 
         public override string ToString()
         {
-            return ToString(format: null, formatProvider: null);
+            return $"0x{m_value:X4}";
+            // return ToString(format: null, formatProvider: null);
+            // TODO: Implement this
         }
 
         public string ToString(string format = null, IFormatProvider formatProvider = null)
         {
-            throw new NotImplementedException();
+            return $"0x{m_value:X4}";
+            // throw new NotImplementedException();
+            // TODO: Implement this
         }
 
         public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan<char> format, IFormatProvider formatProvider)
         {
             throw new NotImplementedException();
         }
+
+        // -----------------------Start of to-half conversions-------------------------
+
+        public static implicit operator Half(int value)
+        {
+            bool sign = value < 0;
+            Half h = (uint)(sign ? -value : value); // Math.Abs but doesn't throw exception, because we cast it to uint anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(uint value)
+        {
+            int shiftDist = BitOperations.LeadingZeroCount(value) - 21;
+            if (shiftDist >= 0)
+            {
+                return value != 0 ?
+                    new Half(false, (ushort)(0x18 - shiftDist), (ushort)(value << shiftDist)) :
+                    default;
+            }
+
+            shiftDist += 4;
+            uint sig = shiftDist < 0 ? Ieee754Helpers.ShiftRightJam(value, -shiftDist) : value << shiftDist;
+            return new Half(RoundPackToHalf(false, (short)(0x1C - shiftDist), (ushort)sig));
+        }
+
+        public static implicit operator Half(long value)
+        {
+            bool sign = value < 0;
+            Half h = (ulong)(sign ? -value : value); // Math.Abs but doesn't throw exception, because we cast it to ulong anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(ulong value)
+        {
+            int shiftDist = BitOperations.LeadingZeroCount(value) - 53;
```

Comment:
Just a note: other than the `shiftDist`, this and the `uint` handling is identical.

Reply:
Guess I could just redirect the int->Half conversion to long->Half conversion operator, yeah. Would it be worth changing?

Reply:
...Or alternatively, redirect the long->Half conversion to int->Half. Max/MinValue is 65504/-65504 anyway, so....

Reply:
I think it's just a case were we could share code in the future. It doesn't need to be done now.